internal/characters/tighnari/cons.go

```diff
@@ -0,0 +1,64 @@
+package tighnari
+
+import (
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+// Tighnari's Charged Attack CRIT Rate is increased by 15%.
+func (c *char) c1() {
+	m := make([]float64, attributes.EndStatType)
+	m[attributes.CR] = 0.15
+	c.AddAttackMod(character.AttackMod{
+		Base: modifier.NewBase("tighnari-c1", -1),
+		Amount: func(atk *combat.AttackEvent, t combat.Target) ([]float64, bool) {
+			if atk.Info.AttackTag != combat.AttackTagExtra {
+				return nil, false
+			}
+			return m, true
+		},
+	})
+}
+
+// When there are opponents within the Vijnana-Khanda Field created by Vijnana-Phala Mine, Tighnari gains 20% Dendro DMG Bonus.
+// The effect will last up to 6s if the field's duration ends or if it no longer has opponents within it.
+func (c *char) c2() {
+	// crutch
+	m := make([]float64, attributes.EndStatType)
+	m[attributes.DendroP] = .2
+	c.AddStatMod(character.StatMod{
+		Base:         modifier.NewBase("tighnari-c2", 18*60-skillHitmark), // 12+6
```

Comment:
the field lasts for 8s, the 12s is for his 3 Suffusion stacks (shorter CA firing time)